From: Julian Ospald <julian.ospald@googlemail.com>
Date: Tue Apr  3 03:33:30 UTC 2012

make path_share look explicitly in SHAREDIR instead of potentially-broken
relative path

--- src/porting.cpp
+++ src/porting.cpp
@@ -244,7 +244,7 @@
 	
 	pathRemoveFile(buf, '/');
 
-	path_share = std::string(buf) + "/../share/" + PROJECT_NAME;
+	path_share = std::string(SHAREDIR);
 	//path_share = std::string(INSTALL_PREFIX) + "/share/" + PROJECT_NAME;
 	if (!fs::PathExists(path_share)) {
 		dstream<<"WARNING: system-wide share not found at \""<<path_share<<"\"";
--- src/cmake_config.h.in
+++ src/cmake_config.h.in
@@ -4,6 +4,7 @@
 #define CMAKE_CONFIG_H
 
 #define CMAKE_PROJECT_NAME "@PROJECT_NAME@"
+#define SHAREDIR "@SHAREDIR@"
 #define CMAKE_INSTALL_PREFIX "@CMAKE_INSTALL_PREFIX@"
 #define CMAKE_VERSION_STRING "@VERSION_STRING@"
 #ifdef NDEBUG
